#!/usr/bin/env python
PACKAGE = "mambo_driver"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()
gen.add("cmd_timeout_sec", double_t, 0,
    "Timeout for repeatedly sending takeoff/land/estop commands (sec)",
    5.0, 0.01, 10.0)
gen.add("sleep_quanta_sec", double_t, 0,
    "Internal operations - base unit to sleep in-between processing BLE commands [AFFECTS TELEMETRY_RATE_HZ] (sec)",
    0.1, 0.001, 0.2)
gen.add("max_vert_speed_mps", double_t, 0,
    "Constraint on maximum vertical speed (m/s)",
    0.5, 0.5, 2.0)
gen.add("max_tilt_deg", double_t, 0,
    "Constraint on maximum lateral tilt (deg)",
    5.0, 5.0, 90.0)
gen.add("banked_turn_mode", bool_t, 0,
    "Banked turn mode", False)

pilot_mode_enum = gen.enum([	gen.const("Full_stabilization", str_t, "easy", "Stabilized"),
								gen.const("Vertical_stabilization", str_t, "medium", "Vertical stabilization"),
								gen.const("No_stabilization", str_t, "difficult", "No stabilization")],
								"Set piloting mode")
gen.add("preferred_pilot_mode", str_t, 0, "Enum to change piloting mode, choosing from \"easy\", \"medium\", \"difficult\".", "medium", edit_method = pilot_mode_enum)

exit(gen.generate(PACKAGE, "mambo_driver_node.py", "Mambo"))
